hack/dockerfiles/govulncheck.Dockerfile

```diff
@@ -0,0 +1,32 @@
+# syntax=docker/dockerfile:1
+
+ARG GO_VERSION=1.21.13
+ARG GOVULNCHECK_VERSION=v1.1.3
+ARG FORMAT=text
+
+FROM golang:${GO_VERSION}-alpine AS base
+WORKDIR /go/src/github.com/docker/docker
+RUN apk add --no-cache jq moreutils
+ARG GOVULNCHECK_VERSION
+RUN --mount=type=cache,target=/root/.cache \
+    --mount=type=cache,target=/go/pkg/mod \
+    go install golang.org/x/vuln/cmd/govulncheck@$GOVULNCHECK_VERSION
+
+FROM base AS run
+ARG FORMAT
+RUN --mount=type=bind,target=.,rw <<EOT
+  set -ex
+  mkdir /out
+  ln -s vendor.mod go.mod
+  ln -s vendor.sum go.sum
+  govulncheck -format ${FORMAT} ./... | tee /out/govulncheck.out
+  if [ "${FORMAT}" = "sarif" ]; then
+    # Make sure "results" field is defined in SARIF output otherwise GitHub Code Scanning
+    # will fail when uploading report with "Invalid SARIF. Missing 'results' array in run."
+    # Relates to https://github.com/golang/vuln/blob/ffdef74cc44d7eb71931d8d414c478b966812488/internal/sarif/sarif.go#L69
+    jq '(.runs[] | select(.results == null) | .results) |= []' /out/govulncheck.out | tee >(sponge /out/govulncheck.out)
+  fi
```

Comment:
> Hm...
> 
> ```
> Uploading results
>   Processing sarif files: ["./build/govulncheck.out"]
>   Validating ./build/govulncheck.out
>   Combining SARIF files using the CodeQL CLI
>   Adding fingerprints to SARIF file. See https://docs.github.com/en/enterprise-cloud@latest/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning#providing-data-to-track-code-scanning-alerts-across-runs for more information.
>   Error: Invalid SARIF. Missing 'results' array in run.
> ```

Yes seems govulncheck omitempty for results: https://github.com/golang/vuln/blob/ffdef74cc44d7eb71931d8d414c478b966812488/internal/sarif/sarif.go#L69

So as a workaround set it if not defined to make GitHub happy. Not yet sure if this is something that needs to be fixed on GitHub or govulncheck side.